EIPS/eip-5143.md

```diff
@@ -0,0 +1,224 @@
+---
+eip: 5143
+title: Slippage Protection for Tokenized Vault
+description: An extension of the ERC-4626 standard for improved EOA interactions.
+author: Hadrien Croubois (@amxx)
+discussions-to: https://ethereum-magicians.org/t/eip-5143-slippage-protection-for-tokenized-vaults/9554
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-06-09
+requires: 4626
+---
+
+## Abstract
+
+The following standard extends the [ERC-4626](./eip-4626.md) Tokenized Vault standard with functions dedicated to the safe interaction between EOAs and the vault when price is subject to slippage.
+
+## Motivation
+
+[ERC-4626](./eip-4626.md) security considerations section states that:
+> "If implementors intend to support EOA account access directly, they should consider adding an additional function call for deposit/mint/withdraw/redeem with the means to accommodate slippage loss or unexpected deposit/withdrawal limits, since they have no other means to revert the transaction if the exact output amount is not achieved."
+
+Yet, ERC-4626 does not standardize the corresponding function signatures and behaviors. For improved interroperability, and better support by wallets, it is essential that this optional functions are also standardized.
+
+## Specification
+
+This ERC is an extenson of ERC-4626. Any contract implementing it MUST also implement ERC-4626.
+
+### Methods
+
+#### deposit
+
+Overloaded version of ERC-4626's `deposit`.
+
+Mints `shares` Vault shares to `receiver` by depositing exactly `assets` of underlying tokens.
+
+MUST emit the `Deposit` event.
+
+MUST support ERC-20 `approve` / `transferFrom` on `asset` as a deposit flow.
```

Comment:
EIPs must be linked the first time they are mentioned:

```suggestion
MUST support [EIP-20](./eip-20.md) `approve` / `transferFrom` on `asset` as a deposit flow.
```

---

Since it sounds like reading EIP-20 to understand this EIP is mandatory, I'd recommend putting it in the `requires:` preamble field.

Reply:
This was copied from EIP-4626. Should the `requires` preamble field include nested dependency?